app/src/main/java/zegreanu/cristi/criminalintent/CrimePagerActivity.kt

```diff
@@ -0,0 +1,37 @@
+package zegreanu.cristi.criminalintent
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.viewpager.widget.ViewPager
+import java.util.UUID
+
+class CrimePagerActivity : AppCompatActivity() {
+    private lateinit var viewPager: ViewPager
+    private var crimes = arrayListOf<Crime>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_crime_pager)
+
+        val a = findViewById<ViewPager>(R.id.crime_view_pager)
+        viewPager = a
+        val crimeId = intent.getSerializableExtra(EXTRA_CRIME_ID)
+        crimes = CrimeLab.crimes
+
+        viewPager.adapter = CrimeSlidePagerAdapter(crimes, supportFragmentManager)
+        viewPager.currentItem = crimes.indexOf(CrimeLab[crimeId as UUID])
```

Comment:
Saw this before, and maybe it's worth having an extension on intent, called getUUID(key: String), and do the cast inside